### DISCUSSION TITLE
Training error: KeyError: ‘avg_G_loss’

### DISCUSSION THREAD

**Discussion by xireli** [coqui-ai/TTS#950] [General Q&A]
when i training Melgan vocoder, i got this error:
–> STEP: 34/57 – GLOBAL_STEP: 25000
| > G_stft_loss_mg: 0.85319
| > G_stft_loss_sc: 0.53163
| > G_subband_stft_loss_mg: 0.73830
| > G_subband_stft_loss_sc: 0.56096
| > G_mse_fake_loss: 0.42901
| > G_gen_loss: 1.34204
| > G_adv_loss: 1.07253
| > loss_0: 2.41457
| > grad_norm_0: 0.00000
| > D_mse_gan_loss: 0.30683
| > D_mse_gan_real_loss: 0.06173
| > D_mse_gan_fake_loss: 0.04801
| > loss_1: 0.30683
| > grad_norm_1: 0.00000
| > current_lr_0: 0.0001
| > current_lr_1: 0.0001
| > step_time: 0.1669
| > loader_time: 0.0023
Traceback (most recent call last):
File “/home/code/TTS-main/TTS/trainer.py”, line 813, in fit
self._fit()
File “/home/code/TTS-main/TTS/trainer.py”, line 799, in _fit
self.train_epoch()
File “/home/code/TTS-main/TTS/trainer.py”, line 668, in train_epoch
_, _ = self.train_step(batch, batch_num_steps, cur_step, loader_start_time)
File “/home/code/TTS-main/TTS/trainer.py”, line 626, in train_step
loss_dict[self.config.target_loss] if “target_loss” in self.config else loss_dict[“loss”]
KeyError: ‘avg_G_loss’

every 5k steps save the best model. anyone one can help me? :sob::sob:

**Comment by erogol** [comment]
can you check what the `target_loss` field is in your config.json ?

**Reply by xireli** [reply]
this is the my  [config file](https://github.com/coqui-ai/TTS-recipes/blob/master/LJSpeech/DoubleDecoderConsistency/vocoder_config.json),

**Reply by erogol** [reply]
```"target_loss": "avg_G_loss",  // loss value to pick the best model to save after each epoch```

set this to "G_avg_loss"
